### PR TITLE
[refactor](file-system)(step-1) refactor file sysmte on BE and remove storage_backend

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

See #17764 for details
I have tested:
- Unit test for local/s3/hdfs/broker file system: be/test/io/fs/file_system_test.cpp
- Outfile to local/s3/hdfs/broker.
- Load from local/s3/hdfs/broker.
- Query file on local/s3/hdfs/broker file system, with table value function and catalog.
- Backup/Restore with local/s3/hdfs/broker file system

Not test:
- cold & host data separation case.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run p0

File: be/src/io/fs/buffered_reader.cpp [hunk 1]
```diff
@@ -67,8 +68,7 @@ Status BufferedFileStreamReader::read_bytes(const uint8_t** buf, uint64_t offset
     while (has_read < to_read) {
         size_t loop_read = 0;
         Slice resutl(_buf.get() + buf_remaining + has_read, to_read - has_read);
-        IOContext io_context;
-        RETURN_IF_ERROR(_file->read_at(_buf_end_offset + has_read, resutl, io_context, &loop_read));
+        RETURN_IF_ERROR(_file->read_at(_buf_end_offset + has_read, resutl, &loop_read));
```
Comment (zxealous) [inline be/src/io/fs/buffered_reader.cpp:71]: Why not pass IOContext to function read_at()?  I understand that we use func BufferedFileStreamReader::read_ bytes() to read data, may be we can save IOContext to BufferedFileStreamReader when calling ScalarColumnReader:: init method?

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

File: be/src/io/fs/buffered_reader.cpp (hunk 1 above)
Comment (morningman) [reply be/src/io/fs/buffered_reader.cpp:71]: Here I just removed unused `IOContext`, not modify the origin logic.
Maybe we can add `IOContext` in `ScalarColumnReader` in next PR.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.77 seconds
 stream load tsv:          475 seconds loaded 74807831229 Bytes, about 150 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          74 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230318031246_clickbench_pr_116282.html

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**